File: app/services/translator.js
import axios from "axios";

const API_URL = "http://127.0.0.1:8000/translator/";

export const translateText = async (text, language) => {
  try {
    const response = await axios.post(API_URL, {
      text,
      language,
    });

    return response.data.translated_text;
  } catch (error) {
    console.error(error);
    return text;
  }
};

File: app/routers/translator.py
from fastapi import APIRouter

from app.schemas.translator import TranslateRequest, TranslateResponse
from app.services.translator_service import translate_text

router = APIRouter(
    prefix="/translator",
    tags=["Translator"]
)


@router.post("/", response_model=TranslateResponse)
def translate(request: TranslateRequest):

    translated = translate_text(
        request.text,
        request.language
    )

    return TranslateResponse(
        translated_text=translated
    )